tmol/score/disulfide/potentials/potentials.hh

```diff
@@ -0,0 +1,209 @@
+#pragma once
+
+#include <cmath>
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include <pybind11/pybind11.h>
+
+#include <tmol/score/common/geom.hh>
+#include <tmol/score/common/tuple.hh>
+#include <tmol/score/common/tuple_operators.hh>
+
+#include <tmol/utility/tensor/TensorAccessor.h>
+#include <tmol/numeric/bspline_compiled/bspline.hh>
+
+namespace tmol {
+namespace score {
+namespace disulfide {
+namespace potentials {
+
+using namespace tmol::score::common;
+
+template <typename Real, tmol::Device D>
+TMOL_DEVICE_FUNC void disulfide_potential(
+    const TensorAccessor<Vec<Real, 3>, 1, D> &coords,
+    int block1_CA_ind,
+    int block1_CB_ind,
+    int block1_S_ind,
+    int block2_S_ind,
+    int block2_CB_ind,
+    int block2_CA_ind,
+
+    const DisulfideGlobalParams<Real> &params,
+
+    Real &pose_V,
+    TensorAccessor<Vec<Real, 3>, 1, D> pose_dV_dx) {
+  auto block1_CA = coords[block1_CA_ind];
+  auto block1_CB = coords[block1_CB_ind];
+  auto block1_S = coords[block1_S_ind];
+
+  auto block2_S = coords[block2_S_ind];
+  auto block2_CB = coords[block2_CB_ind];
+  auto block2_CA = coords[block2_CA_ind];
+
+  auto ssdist = distance<Real>::V_dV(block1_S, block2_S);
+  auto csang_1 = pt_interior_angle<Real>::V_dV(block1_CB, block1_S, block2_S);
+  auto csang_2 = pt_interior_angle<Real>::V_dV(block2_CB, block2_S, block1_S);
+  auto dihed =
+      dihedral_angle<Real>::V_dV(block1_CB, block1_S, block2_S, block2_CB);
+  auto disulf_ca_dihedral_angle_1 =
+      dihedral_angle<Real>::V_dV(block1_CA, block1_CB, block1_S, block2_S);
+  auto disulf_ca_dihedral_angle_2 =
+      dihedral_angle<Real>::V_dV(block2_CA, block2_CB, block2_S, block1_S);
+
+  const Real MEST = exp(-20.0);
+  const Real WT_DIH_SS(0.1);
+  const Real WT_DIH_CS(0.1);
+  const Real WT_ANG(0.1);
+  const Real WT_LEN(0.1);
+  const Real SHIFT(2.0);
+
+  const Real res1_d_multiplier(1.0);
+  const Real res2_d_multiplier(1.0);
```

Comment:
Rosetta has considerations for different chiralities. I did not include that functionality, but I left these variables in here. Should I remove them, or should we leave them and add that code later?